python/cuda_parallel/cuda/parallel/experimental/iterators/_iterators.py

```diff
@@ -97,46 +96,50 @@ def __init__(
           A numba type of the iterator state.
         value_type
           The numba type of the value returned by the dereference operation.
-        prefix
-          An optional prefix added to the iterator's methods to prevent name collisions.
         """
         self.cvalue = cvalue
         self.numba_type = numba_type
         self.state_type = state_type
         self.value_type = value_type
-        self.prefix = prefix
         self.kind_ = self.__class__.iterator_kind_type(self.value_type, self.state_type)
         self.state_ = IteratorState(self.cvalue)
+        self._ltoirs: Dict[str, bytes] | None = None
 
     @property
     def kind(self):
         return self.kind_
 
-    # TODO: should we cache this? Current docs environment doesn't allow
-    # using Python > 3.7. We could use a hand-rolled cached_property if
-    # needed.
     @property
     def ltoirs(self) -> Dict[str, bytes]:
-        abi_suffix = _get_abi_suffix(self.kind)
-        advance_abi_name = f"{self.prefix}advance_{abi_suffix}"
-        deref_abi_name = f"{self.prefix}dereference_{abi_suffix}"
-        advance_ltoir, _ = cached_compile(
-            self.__class__.advance,
-            (
-                self.numba_type,
-                types.uint64,  # distance type
-            ),
-            output="ltoir",
-            abi_name=advance_abi_name,
-        )
+        if self._ltoirs is None:
+            advance_abi_name = "advance_" + _get_abi_suffix(self.kind)
+            deref_abi_name = "dereference_" + _get_abi_suffix(self.kind)
```

Comment:
I changed these line in gh-4325 to 

```
        abi_suffix = _get_abi_suffix(self.kind)
        advance_abi_name = f"{self.prefix}advance_{abi_suffix}"
        deref_abi_name = f"{self.prefix}dereference_{abi_suffix}"
```

My reasoning was to avoid repeated calls to `_get_abi_suffix(self.kind)`. Since the prefix is now gone, perhaps it should be 

 ```suggestion
             abi_suffix = _get_abi_suffix(self.kind)
             advance_abi_name = f"advance_{abi_suffix}"
             deref_abi_name = f"dereference_{abi_suffix}"
   ```